test/puppetlabs/trapperkeeper/services/authorization/authorization_service_test.clj

```diff
@@ -89,9 +137,22 @@
             {:keys [status body]} (app req)]
         (is (= status 200))
         (is (= body "Prefix: !dlroW olleH")))))
+  (testing "(TK-260) Authorizing unauthenticated requests"
```

Comment:
Might be good to throw another test in here somewhere which validates that that authenticated request - one that should also be allowed by rule - will still end up being allowed for a rule which has `allow_authenticated: true`.  I think that will work fine with the changes as you have them now.  The extra test might help ensure that we don't inadvertently break that when we make changes around this stuff later.